### PR TITLE
Fix data migrations

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...

- missing plugin permission in data migrations
- use `iterator()` instead of `all()`
- make `name` in pluginUpdate query required 

# Impact

* [ ] New migrations
* [X] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [X] Privileged queries and mutations are guarded by proper permission checks
* [X] Database queries are optimized and the number of queries is constant
* [X] Database migration files are up to date
* [X] The changes are tested
* [X] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/schema.graphql [hunk 1]
```diff
@@ -2424,7 +2424,7 @@ type Mutation {
   giftCardCreate(input: GiftCardCreateInput!): GiftCardCreate
   giftCardDeactivate(id: ID!): GiftCardDeactivate
   giftCardUpdate(id: ID!, input: GiftCardUpdateInput!): GiftCardUpdate
-  pluginUpdate(id: ID!, input: PluginUpdateInput!): PluginUpdate
+  pluginUpdate(id: ID!, input: PluginUpdateInput!, name: String!): PluginUpdate
```
Comment (maarcingebala) [inline saleor/graphql/schema.graphql:2427]: `name` should be rather part of `PluginUpdateInput`.

File: saleor/account/migrations/0041_permissions_to_groups.py [hunk 2]
```diff
@@ -39,11 +39,8 @@ def get_counter_value(Group):
 def create_permissions_mapping(User):
     """Create mapping permissions to users and potential new group name."""
     mapping = defaultdict(set)
-    users = (
-        User.objects.filter(user_permissions__isnull=False)
-        .distinct()
-        .prefetch_related("user_permissions")
-    )
+    users = User.objects.filter(user_permissions__isnull=False).distinct().iterator()
```
Comment (fowczarek) [inline saleor/account/migrations/0041_permissions_to_groups.py:42]: Why we use `distinct()` here?

Comment (IKarbowiak) [reply saleor/account/migrations/0041_permissions_to_groups.py:42]: Because it returns duplicated items when a user has more than one `user_permissions`

Comment (korycins) [reply saleor/account/migrations/0041_permissions_to_groups.py:42]: 👮‍♀️

**Pull request merged**